lib/daru/io/importers/json.rb

```diff
@@ -0,0 +1,174 @@
+        #   #=>        13 (2015 HF11 2015-04-17       19.2      34.89      0.225       Amor       2.99 ...
+        #   #=>        14 (2015 GK50 2015-04-05       20.5      19.07      0.237       Amor       5.39 ...
+        #   #=>       ...        ...        ...        ...        ...        ...        ...        ... ...
+        #
+        # @example Importing from remote JSON response with x-path selectors
+        #
+        #   url = 'http://api.tvmaze.com/singlesearch/shows?q=game-of-thrones&embed=episodes'
+        #   df  = Daru::IO::Importers::JSON.new(url,
+        #             "$.._embedded..episodes..name",
+        #           "$.._embedded..episodes..season",
+        #           "$.._embedded..episodes..number",
+        #            index: (10..70).to_a,
+        #            RunTime: "$.._embedded..episodes..runtime"
+        #         ).call
+        #
+        #
+        #   # Note that the hash x-path selectors like `index`, `RunTime`, etc.
+        #   # should be given after normal x-path selectors like that of `name`,
+        #   # `season` and `number`.
+        #
+        #   df
+        #
+        #   #=> #<Daru::DataFrame(61x4)>
+        #   #=>               name     season     number    RunTime
+        #   #=>      10 Winter is           1          1         60
+        #   #=>      11 The Kingsr          1          2         60
+        #   #=>      12  Lord Snow          1          3         60
+        #   #=>      13 Cripples,           1          4         60
+        #   #=>      14 The Wolf a          1          5         60
+        #   #=>      15 A Golden C          1          6         60
+        #   #=>      16 You Win or          1          7         60
+        #   #=>      17 The Pointy          1          8         60
+        #   #=>      18     Baelor          1          9         60
+        #   #=>      19 Fire and B          1         10         60
+        #   #=>      20 The North           2          1         60
+        #   #=>      21 The Night           2          2         60
+        #   #=>      22 What is De          2          3         60
+        #   #=>      23 Garden of           2          4         60
+        #   #=>      24 The Ghost           2          5         60
+        #   #=>     ...        ...        ...        ...        ...
+        def initialize(input, *arrays, order: nil, index: nil, **hashes)
+          super(binding)
+          @data       = []
+          @auto_order = []
+        end
+
+        def call
+          json = read_json
+
+          if @hashes.empty?
+            if @arrays.size == 1
+              @data = get_xpath(json, @arrays.first)
+            elsif @arrays.empty?
+              parse_without_hash json
+            else
+              parse_with_array json
+            end
+          else
+            parse_with_array json unless @arrays.empty?
+            parse_with_hash json
+          end
```

Comment:
The same thing about nested `if`s, and several other concerns here. 
For example, names like `_with_hash` and `_with_array` are not really communicative.
What, in fact, we should have here is:
1. independent step: parse/assing index, if provided
2. independent step: parse/assign column names
3. independent step: parse/assign all the columns, being them named or unnamed

So, I'd prefer to see code written in this terms, like (pseudocode):
```ruby
index = jsonpath(@index)
order = jsonpath(@order)
columns = parse_columns 
DataFrame.new(columns, index: index, order: order)

# the #columns probably uses @columns, @named_columns a way like this:
return jsonpath(@columns.first) if @columns.count == 1 && @named_columns.empty?
@columns.map { |path| Vector.new(jsonpath(path)) } + 
  @named_columns.map { |name, path| Vector.new(jsonpath(path), name: name) }
```

The only thing that is kinda tricky here, is what if the user provided both `order` param and named columns selectors.
I believe that this case should be explicitly prohibited, as it is almost impossible to guess what user have really wanted.